### PR TITLE
Add optional branch name regexp check

### PR DESCRIPTION
**Pull request by seqizz**
I am not too happy code-wise since I had to duplicate/modify some stuff, but as far as I can test this one works.
The check is optional like others, enabled when `.igcommit-branch-name.conf` file is found on the repository root.

### CODE REVIEW

File: igcommit/commit_list_checks.py [hunk 1]
```diff
@@ -253,3 +256,94 @@ def check_contributor(self, contributor, commit):
...
+                'for branch name comparison:\n{}'
+                .format("\n".join(broken_regexes))
+            )
+        if not self.check_branch_name(valid_regexes, branch_name):
+            yield (
+                Severity.ERROR,
+                'branch name does not match any of the '
+                'alowed regexp patterns:\n{}'
+                .format("\n".join(valid_regexes))
+            )
```
Comment (kofrezo) [inline igcommit/commit_list_checks.py:349]: How about:
```
yield (
    Severity.ERROR,
    "{branch_name} is not an allowed branch name.\n"
    "Branch names must match any of the follow regular expressions:\n"
    "{valid_regexes.join('\n-')}"
)
```

File: igcommit/commit_list_checks.py [hunk 2]
```diff
@@ -253,3 +256,94 @@ def check_contributor(self, contributor, commit):
...
+        """Update used configuration files, return true if any exist
+
+        This is copied from file_checks.py and slightly modified.
+        Please check the original file for explanation and
+        shortcomings.
+        """
+        config_exists = False
+        for config_file in self.config_files:
+            prev_commit = config_file.commit
+            # We will check if the config file exist on the latest commit
```
Comment (kofrezo) [inline igcommit/commit_list_checks.py:287]:
```suggestion
            # We will check if the config file exists on the latest commit
```

**Review by kofrezo — APPROVED** [review]
The preparation of the configuration files is a bit black magic to me so I may have overseen something but overall it looks good to me. 

See comments for minor improvements you could do (no blockers tough).

File: igcommit/commit_list_checks.py [hunk 3]
```diff
@@ -253,3 +256,94 @@ def check_contributor(self, contributor, commit):
...
+    def get_problems(self):
+        branch_name = self.commit_list.branch_name
+        valid_regexes, broken_regexes = self.get_allowed_regexes()
+        if broken_regexes:
+            yield (
+                Severity.WARNING,
+                'following string is not a valid pattern '
+                'for branch name comparison:\n{}'
+                .format("\n".join(broken_regexes))
+            )
```
[the remark this answers is not in the document]
Comment (seqizz) [reply igcommit/commit_list_checks.py:342]: Due to the way configuration-needed checks work (they all look into the repository), I am also reading the config file which comes via latest commit. So pushing user is likely causing the issue & can fix by changing the regex in the config file.
That also means people can get rid of the restrictions by removing/editing the file, but this is not different from adding a `disable` line to pylintrc to pass a warning :shrug:

File: igcommit/commit_list_checks.py (hunk 1 above)
Comment (seqizz) [reply igcommit/commit_list_checks.py:349]: I refrained from f-strings like other functions, looks like we still support Python3.4. But I'll do the wording better :+1:

**Pull request merged**